### PR TITLE
Release v1.0.0 alpha.1

### PR DESCRIPTION
**Pull request by xofyarg**
Bump the version, add several minor fixes.

### CODE REVIEW

File: tests/test-vectors.json [hunk 1]
```diff
@@ -1 +1 @@
...22bb93dd4546c0566ec2b8fde5e8f3ae48560341fbd8a8ca970ae8a948cfe9e180992c0c4859d1d7ae2c633364f355a4061a2c65f91bf5232f11a8d53ce22ed63db2c40453ef3d6dd7116f8896","obliviousResponse":"020010f849b229ba51466ccb50d4dec310756301e8146a7de2ed1428eb2bcd904ac550f519e3031f49afa80a53ffbab45ffb599da99b03278dd056474d92a3c03c79279a3390ef15e17192abe46dfa3215b4495fe412db9929ce48296c7419aea3e9370d7ed649967f7ff250b8e99fe1b18802135ef6f8435af2ab7aeba58a25d2734249552f0fc10543f74b606431e095fa919991aa7910592bc17cb69c55ee49efd784118a1de15639762c167f50f8ff714d4bea13256f2857537977e24cc7552490a64dbcba2a6042a4688110365963671a71615a3ba93f5bf097d292449b89580fb01c2301b7ffc1359e382c92376e7cb95f2fc6c137888ca463d40a248186b1e5296a6ac419c65b331b9581a2377aa06b4f2b65811caba7a8e1452641524982ed4eda5d7747369aa2a5a4e2a7824bfd43f542f2f1024367006aedbb53705d451b0132326e3221040a4e78f404f177ee8a9dc62efd28f579c6850327ae958bb24008b2523dc6adacfe07d70f409af652000233ac6fcff02a76b1238575a3563b4b2eb7c29a3de1863b7fb402f9d0399e2912bc1427b9b392a0c44722cee8ee46e3cf75cc4032f159b67753168a3341e1b220e20f55a494d7b9e92453186a24dbf2be9f994828823496feaeed0f5b3cdbdecdd763a25fc460a5735a4209a4fd288eb1b8173978cefb2085ac61dd73bfa79bb21beb8b3f22b559af7e"}]}]
```
Comment (chris-wood) [inline tests/test-vectors.json:1]: Were these manually updated?

File: src/protocol.rs [hunk 2]
```diff
@@ -864,4 +892,13 @@ mod tests {
...
         }
     }
+
+    #[test]
+    fn padding() {
+        let query = ObliviousDoHMessagePlaintext::new(&[], 0);
+        assert_eq!(query.padding_len(), 0);
+
+        let query = ObliviousDoHMessagePlaintext::new(&[], 2);
+        assert_eq!(query.padding_len(), 2);
```
Comment (chris-wood) [inline src/protocol.rs:926]: Could we add tests that serialize a message and check that there are the appropriate amount of padding bytes, all zero, too?

**Review by chris-wood — COMMENTED** [review]
This LGTM, modulo one test suggestion and one question about the vectors.

File: tests/test-vectors.json (hunk 1 above)
Comment (xofyarg) [reply tests/test-vectors.json:1]: Yes.

File: src/protocol.rs (hunk 2 above)
Comment (xofyarg) [reply src/protocol.rs:926]: Added a test to cover the non-zero padding check logic.

File: src/protocol.rs [hunk 3]
```diff
@@ -101,6 +101,30 @@ pub enum Error {
     Internal,
 }
 
+impl From<HpkeError> for Error {
+    fn from(e: HpkeError) -> Self {
+        Self::Hpke(HpkeErrorWrapper(e))
+    }
+}
+
+/// This is a wrapper for HpkeError, as the type doesn't support Eq.
```
Comment (vavrusa) [inline src/protocol.rs:110]: Maybe try open an upstream PR? Adding PartialEq or kind() method seems reasonable.

Comment (xofyarg) [reply src/protocol.rs:110]: Yeah, it's in the plan: https://github.com/rozbb/rust-hpke/pull/13

**Review by vavrusa — APPROVED** [review]

File: tests/test-vectors.json (hunk 1 above)
Comment (chris-wood) [reply tests/test-vectors.json:1]: Let's use the test vectors from here: https://github.com/cloudflare/odoh-go/pull/12

**Review by chris-wood — CHANGES_REQUESTED** [review]
Requesting changes to the test vectors to ensure we interop, but otherwise LGTM.

**Review by chris-wood — APPROVED** [review]
